hoomd/hpmc/ShapeUnion.h

```diff
@@ -114,64 +110,90 @@ struct union_params : param_base
         for (unsigned int i = 0; i < mparams.size(); ++i)
             mparams[i].set_memory_hint();
         }
+
     #endif
 
-    //! Shape constructor
-    union_params(unsigned int _N, bool _managed)
-        : N(_N)
+    #ifndef __HIPCC__
+
+    /** Construct with a given number of members
+
+        @note Use of this constructor with N != 0 is intended only for unit tests
+    */
+    DEVICE ShapeUnionParams(unsigned int _N)
+        : mpos(_N, false), morientation(_N, false), mparams(_N, false),
```

Comment:
I know that this from previous code, but maybe we should make a note to name this `m_pos` etc. to keep consistent styling.

Reply:
Added comment d98589281